src/NzbDrone.Core/Download/CompletedDownloadService.cs

```diff
@@ -115,6 +120,7 @@ private void Import(TrackedDownload trackedDownload)
             {
                 trackedDownload.State = TrackedDownloadStage.Imported;
                 _eventAggregator.PublishEvent(new DownloadCompletedEvent(trackedDownload));
+                downloadClient.SetImportedCategory(trackedDownload.DownloadItem);
```

Comment:
Architecturally, CompletedDownloadService doesn't know about download clients.
Might be good idea to move it to DownloadEventHub, which is tasked to handle the DownloadCompletedEvent (amongst others), and already has access to the download client in order to remove downloads.
It's a fairly trivial change, all pieces are already in place.

Reply:
I'm assuming you mean within DownloadCompletedEvent... does this method imply the release has been imported already?

Reply:
No, within the [event handler in DownloadEventHub](https://github.com/Sonarr/Sonarr/blob/phantom-develop/src/NzbDrone.Core/Download/DownloadEventHub.cs#L38-L46).
DownloadCompletedEvent is a 'message' that can be raised by CompletedDownloadService and ManualImportService (Manual/Forced Import).